torch/cuda/memory.py

```diff
@@ -8,13 +8,12 @@
 import sys
 import warnings
 from inspect import signature
-from typing import Any, Literal, Optional, Union
+from typing import Any, Literal, Optional
 from typing_extensions import deprecated
 
 import torch
 from torch import _C
 from torch._utils import _dummy_type
-from torch.types import Device
```

Comment:
Ugh, why not just keep this from import? Would make the rest of the changes more concise?

Reply:
Thanks for your review. I keeped this import in the `TYPE_CHECKING` region, does this make sense?